In `tiny_jpeg.h` , `TJEI_FORCE_INLNE` should be `static` on Windows 10, clang but not.

I tryed to compile on windows 10 with clang, cmake and ninja with option `SDL3IMAGE_VENDORED` on. These are the versions I used.
```cmd
> cmake --version
cmake version 3.26.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).

> ninja --version
1.11.0

> clang --version
clang version 15.0.7
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files\LLVM\bin
```
When I try to compile, it generates the following warning.
```cmd
[290/344] Building C object _deps/sdl2_image-build/CMakeFiles/SDL2_image.dir/IMG_jpg.c.obj
In file included from C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/IMG_jpg.c:683:
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:635:9: warning: static function 'tjei_write' is used in an inline function with external linkage [-Wstatic-in-inline]
        tjei_write(state, &c, 1, 1);
        ^
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:615:1: note: use 'static' to give inline function 'tjei_write_bits' internal linkage
TJEI_FORCE_INLINE void tjei_write_bits(TJEState* state,
^
static
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:176:27: note: expanded from macro 'TJEI_FORCE_INLINE'
#define TJEI_FORCE_INLINE __forceinline
                          ^
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:473:13: note: 'tjei_write' declared here
static void tjei_write(TJEState* state, const void* data, size_t num_bytes, size_t num_elements)
            ^
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:639:13: warning: static function 'tjei_write' is used in an inline function with external linkage [-Wstatic-in-inline]
            tjei_write(state, &z, 1, 1);
            ^
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:615:1: note: use 'static' to give inline function 'tjei_write_bits' internal linkage
TJEI_FORCE_INLINE void tjei_write_bits(TJEState* state,
^
static
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:176:27: note: expanded from macro 'TJEI_FORCE_INLINE'
#define TJEI_FORCE_INLINE __forceinline
                          ^
C:/Users/%USERNAME%/Projects/dmk/build/_deps/sdl2_image-src/tiny_jpeg.h:473:13: note: 'tjei_write' declared here
static void tjei_write(TJEState* state, const void* data, size_t num_bytes, size_t num_elements)
            ^
2 warnings generated.
```
NOTE : I chaged my user name to `%USERNAME%`
This seems that clang defines `_MSC_VER` , so the macro settings gets [wrong](https://github.com/libsdl-org/SDL_image/blob/0bd8290b776baf719d9fc7969b39b0a2876bb41b/tiny_jpeg.h#L176). Could you add `!defined(__clang__)` or something?

--- a/tiny_jpeg.h
+++ b/tiny_jpeg.h
@@ -173,10 +173,12 @@ int tje_encode_with_func(tje_write_func* func,
 
 
 #if defined(_MSC_VER)
-#define TJEI_FORCE_INLINE __forceinline
+#define TJEI_FORCE_INLINE static __forceinline
 // #define TJEI_FORCE_INLINE __declspec(noinline)  // For profiling
+#elif ((defined(__GNUC__) && (__GNUC__ >= 4)) || defined(__clang__))
+#define TJEI_FORCE_INLINE  __attribute__((always_inline)) static __inline
 #else
-#define TJEI_FORCE_INLINE static // TODO: equivalent for gcc & clang
+#define TJEI_FORCE_INLINE static __inline
 #endif
 
 // Only use zero for debugging and/or inspection.

